ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/UpdateMutabilityInSignatureFix.fs

```diff
@@ -0,0 +1,71 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+
+type UpdateMutabilityInSignatureFix(error: ValueNotContainedMutabilityAttributesDifferError) =
+    inherit FSharpQuickFixBase()
+
+    let tryFindBindingSignature (implementationDeclaredName: string) (declaredElement: IFSharpMember) =
+        let parentDeclarations = declaredElement.ContainingType.GetDeclarations()
+
+        // Find the parent signature counter part
+        let signatureNameModuleDeclaration =
+            parentDeclarations
+            |> Seq.tryPick (fun d ->
+                match d with
+                | :? INamedModuleDeclaration as signatureModule when
+                    signatureModule.GetSourceFile().IsFSharpSignatureFile -> Some signatureModule
+                | _ -> None)
+
+        signatureNameModuleDeclaration
+        |> Option.bind (fun signatureModule ->
+            signatureModule.Members
+            |> Seq.tryPick (fun sigMember ->
+                match sigMember with
+                | :? IBindingSignature as bindingSignature ->
+                    match bindingSignature.HeadPattern with
+                    | :? ITopReferencePat as bindingNamePat
+                        // This is a bit lucky to work.
+                        // There must be something more clever we can try
+                        when bindingNamePat.DeclaredName = implementationDeclaredName ->
+                        Some bindingSignature
+                    | _ -> None
+                | _ -> None)
+        )
+
+    let tryFindImplementationBindingInfo (pat: ITopReferencePat) =
+        if isNull pat then None else
+        let topLevelBinding = TopBindingNavigator.GetByHeadPattern(pat)
+        if isNull topLevelBinding then None else
+        let declaredElement = topLevelBinding.DeclaredElement :?> IFSharpMember
+        if isNull declaredElement then None else
+        let mfv = declaredElement.Mfv
+        if isNull mfv then None else
+        if not mfv.HasSignatureFile then None else
+        Some (topLevelBinding, declaredElement)
```

Comment:
`TopBindingNavigator.GetByHeadPattern(pat)` won't cover cases like
```fsharp
let (a, b) = ...
```
or
```fsharp
let (i: int) = ...
```
I'm not sure if we want to cover them, but if we do, there's `Binding` property that makes it easier to get a binding through various nested patterns.

Getting a declared element from a binding would only work reliably for a single top-level pattern, so getting it from the initial pattern would be better too.

This is what the change could look like if want to support nested patterns:
```suggestion
        if isNull pat then None else

        match pat.Binding, pat.DeclaredElement.As<IFSharpMember>() with
        | null, _ | _, null -> None
        | binding, fsMember ->

        let mfv = fsMember.Mfv
        if isNull mfv then None else

        Some(binding, fsMember)
```

The `x.ExecutePsiTransaction` would also get simplified if you use `Binding.